lib/Doctrine/DBAL/Schema/Column.php

```diff
@@ -118,9 +119,17 @@ public function setOptions(array $options)
     {
         foreach ($options as $name => $value) {
             $method = "set".$name;
-            if (method_exists($this, $method)) {
-                $this->$method($value);
+            if ( ! method_exists($this, $method)) {
+                // next major: use InvalidArgumentException::fromUnsupportedOption()
+                @trigger_error(sprintf(
```

Comment:
Scrutinizer does not like the silencing

Reply:
It seems like Scrutinizer allows silencing in an `if` block, but that's no option for this case. I haven't worked much with the Scrutinizer config, but maybe there's a way to ignore it just for `trigger_error`.

Reply:
Maybe this link would help:  https://scrutinizer-ci.com/docs/reviews/filtering_issues_and_false_positives ?

Reply:
@lcobucci maybe you can help us here?

Reply:
Could anyone please point me to the article explaining why this silencing operator is used in Symfony and we want it here? As far as I understand, the built-in PHP error handler will not display such errors. A user-defined handler, if it respects `error_reporting()`, it will ignore such errors as well. Does it mean users will have to use some specialized error handler to catch these errors?

Reply:
@morozov  here is the handler: https://github.com/symfony/symfony/blob/master/src/Symfony/Bridge/PhpUnit/DeprecationErrorHandler.php IIRC most implementations ignore silenced errors, this one does not. I think a PSR is coming regarding all this.

Reply:
We want it here not only for Symfony users, but for us: we want to know if we use a deprecated path somewhere else in the lib